public/scripts/views/forms/custom-id.js

```diff
@@ -0,0 +1,164 @@
+(function (window) {
+  "use strict";
+  //note field below field to validation errors
+  //2021-07-25-17-47-59.png
+  //validation rule
+  window.ls.container.get("view").add({
+    selector: "data-custom-id",
+    controller: function (element, sdk, console) {
+      var prevData = "";
+      let idType = element.dataset["id-type"];
+
+      var div = window.document.createElement("div");
+
+      div.className = "input-copy";
+
+      var button = window.document.createElement("i");
+
+      button.type = "button";
+      button.style.cursor = "pointer";
+
+      var writer = document.createElement("input");
+      writer.type = "text";
+      writer.className = "";
+      writer.setAttribute("maxlength", element.getAttribute("maxlength"));
+      var placeholder = element.getAttribute("placeholder");
+      if (placeholder) {
+        writer.setAttribute("placeholder", placeholder);
+      }
+
+      var info = window.document.createElement("div");
+      info.className = "text-fade text-size-xs margin-top-negative-small margin-bottom";
+
+      div.appendChild(writer);
+      div.appendChild(button);
+      element.parentNode.insertBefore(div, element);
+      element.parentNode.insertBefore(info, div.nextSibling);
+
+      var switchType = function (event) {
+        if (idType == "custom") {
+          idType = "auto";
+          setIdType(idType);
+        } else {
+          idType = "custom";
+          setIdType(idType);
+        }
+      }
+
+      var validate = function (event) {
+        var service = element.dataset["validator"];
+        service = service.split('.');
```

Comment:
```suggestion
        const [service, method] = service.split('.');
```
Can we do something like this to prevent accessing by the index?

Reply:
Nice!